### PR TITLE
python310Packages.ipyniivue: init at 1.0.2

### PR DESCRIPTION
**Pull request by bcdarwin**
python310Packages.jupyter-ui-poll: init at 0.2.2

###### Description of changes

Add `python310Packages.ipyniivue`, a package for displaying NIfTI medical imaging data in Jupyter notebooks, and its dependency `python310Packages.jupyter-ui-poll` for handling user input in widgets.

###### Things done

In addition, I checked that the example in the README runs correctly in a Jupyter notebook.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ipyniivue/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  nativeBuildInputs = [ hatchling hatch-jupyter-builder ];
+
+  propagatedBuildInputs = [ ipywidgets jupyter-ui-poll ];
+
+  nativeCheckImports = [ pytestCheckHook ];
+  pythonImportsCheck = [ "ipyniivue" ];
+
+  meta = with lib; {
+    description = "Show a nifti image in a webgl 2.0 canvas within a jupyter notebook cell";
+    homepage = "https://github.com/niivue/ipyniivue";
```
Comment (fabaff) [inline pkgs/development/python-modules/ipyniivue/default.nix:33]: Perhaps the changelog will improve over time.

```suggestion
    homepage = "https://github.com/niivue/ipyniivue";
    changelog = "https://github.com/niivue/ipyniivue/releases/tag/${version}";
```

File: pkgs/development/python-modules/jupyter-ui-poll/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "Kirill888";
+    repo = "jupyter-ui-poll";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-DWZFvzx0aNTmf1x8Rq19OT0PFRxdpKefWYFh8C116Fw";
+  };
+
```
Comment (fabaff) [inline pkgs/development/python-modules/jupyter-ui-poll/default.nix:22]: According to [`setup.cfg`](https://github.com/Kirill888/jupyter-ui-poll/blob/v0.2.2/setup.cfg#L35-L36). Interesting that didn't lead to a failure.

```suggestion
  nativeBuildInputs = [ setuptools ];
```

File: pkgs/development/python-modules/jupyter-ui-poll/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  propagatedBuildInputs = [
+    ipython
+  ];
+
+  doCheck = false;  # no tests in package :(
+  pythonImportsCheck = [ "jupyter_ui_poll" ];
+
+  meta = with lib; {
+    description = "Block jupyter cell execution while interacting with widgets";
+    homepage = "https://github.com/Kirill888/jupyter-ui-poll";
```
Comment (fabaff) [inline pkgs/development/python-modules/jupyter-ui-poll/default.nix:32]:
```suggestion
    homepage = "https://github.com/Kirill888/jupyter-ui-poll";
    changelog = "https://github.com/Kirill888/jupyter-ui-poll/releases/tag/v${version}";
```

File: pkgs/development/python-modules/ipyniivue/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/ipyniivue/default.nix:33]: Added.

File: pkgs/development/python-modules/jupyter-ui-poll/default.nix (hunk 3 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/jupyter-ui-poll/default.nix:32]: Done.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 229346` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.ipyniivue</li>
    <li>python310Packages.ipyniivue.dist</li>
    <li>python310Packages.jupyter-ui-poll</li>
    <li>python310Packages.jupyter-ui-poll.dist</li>
    <li>python311Packages.ipyniivue</li>
    <li>python311Packages.ipyniivue.dist</li>
    <li>python311Packages.jupyter-ui-poll</li>
    <li>python311Packages.jupyter-ui-poll.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Pull request merged**